### PR TITLE
Manurhin MR76

### PR DESCRIPTION
**Pull request by Max-023**
## About The Pull Request
This PR adds two new .357 revolvers to the game for testing! The MR76 and the Major's variant of the MR76, a French autorevolver known for its reliability! Using the same rounds as the similarly chambered SW 357 and CMB, the differences are that the MR76 does not have a burst mode. Instead, it gets a singlefire auto mode! Rather than firing two bullets at a time, you can live our your dream of being a rootin' tootin' shootin' cowboy and show these posers what high noon really means!

## Why It's Good For The Game
Right now, the revolvers are either a meme or a power player's wet dream. The MR76 is not the most powerful revolver in the game and shouldn't be. Doing less damage than the M44 and not having the burst fire other autorevolvers are known for, this one is supposed to appeal to the greater playerbase as a worthy sidearm, or a legendary primary. 1 of the biggest complaints with the M44 was the shakiness of it and the scatter without the stock. The MR76 does not have these issues, being much more controllable. The unique ability to auto fire with the MR76 will surely attract more players as you now can dump what ever is left in your cylinders in a pinch to avoid danger.

On top of this, both the MR76 and the SW 357 get access to a completely new type of ammo for the revolvers based on the marksman ammo for the M44. Sacrificing 5 damage for 10 penetration, you will have an easier time against armored castes who want to underestimate your pistol. Hopefully this could bring light to the revolvers and lead to future development with the unused firearm.

## Changelog
:cl:
add: MR76 and Major's MR76
add: Marksman .357
fix: M44 caliber bug where removing marksman rounds turns them into regular rounds
imageadd: Adds MR76 and Major's MR76 to gun.dmi and the mob item .dmis
server: There is no way to get this item in game so you will need to spawn them in
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/revolvers.dm [hunk 1]
```diff
@@ -444,3 +444,45 @@
...
+
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC)
+	fire_delay = 0.3 SECONDS
+	scatter_unwielded = 20
+	scatter = -20
+	recoil = 0
+	recoil_unwielded = 0
+	damage_mult = 1.25
+
+/obj/item/weapon/gun/revolver/small/unique_action(mob/user)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/revolvers.dm]:
```suggestion
/obj/item/weapon/gun/revolver/mr76/unique_action(mob/user)
```

File: code/modules/projectiles/updated_projectiles/guns/revolvers.dm [hunk 2]
```diff
@@ -444,3 +444,45 @@
...
+						/obj/item/attachable/reddot,
+						/obj/item/attachable/flashlight,
+						/obj/item/attachable/extended_barrel,
+						/obj/item/attachable/compensator,
+						/obj/item/attachable/scope,
+						/obj/item/attachable/lasersight,
+						/obj/item/attachable/scope/mini)
+	attachable_offset = list("muzzle_x" = 32, "muzzle_y" = 22,"rail_x" = 13, "rail_y" = 24, "under_x" = 25, "under_y" = 17, "stock_x" = 25, "stock_y" = 17)
+
+	gun_firemode_list = list(GUN_FIREMODE_SEMIAUTO, GUN_FIREMODE_AUTOMATIC)
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/revolvers.dm]: theres no such thing as an automatic revolver

Comment (Max-023) [reply code/modules/projectiles/updated_projectiles/guns/revolvers.dm]: Double Action revolvers pull the hammer back and push it forward with 1 trigger pull. The full auto mode would just be rapid trigger pulls from the user and QoL so you don't have to get carpel tunnel in order to ahoot quickly.

Comment (Alterist) [reply code/modules/projectiles/updated_projectiles/guns/revolvers.dm]: *cough* https://en.wikipedia.org/wiki/Webley%E2%80%93Fosbery_Automatic_Revolver *cough*

**Comment by Alterist** [comment]
description says lean, but the sprite looks as large as the mateba, not a real issue, just my 2 cents on the sprite

File: code/modules/projectiles/updated_projectiles/guns/revolvers.dm (hunk 2 above)
Comment (Max-023) [reply code/modules/projectiles/updated_projectiles/guns/revolvers.dm]: Oof

**Comment by Max-023** [comment]
I am also redoing all the revolver sprites, so while the M44 sprite is small *now*, it won't be in the future.

File: code/modules/projectiles/updated_projectiles/magazines/revolvers.dm [hunk 3]
```diff
@@ -100,7 +110,9 @@
 /obj/item/ammo_magazine/internal/revolver/small
 	default_ammo = /datum/ammo/bullet/revolver/small
 	caliber = ".357"
-	gun_type = /obj/item/weapon/gun/revolver/small
+	gun_type = list(
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/magazines/revolvers.dm]: The current system doesn't work for lists. This needs a refactor to work.

**Comment by TheBimmer231** [comment]
On hold till I edit the sprites you sent me.

File: code/modules/projectiles/updated_projectiles/magazines/revolvers.dm (hunk 3 above)
Comment (Rohesie) [reply code/modules/projectiles/updated_projectiles/magazines/revolvers.dm]: This hasn't been fixed. Please do not resolve conversations on your own without addressing the issues.

**Comment by TheBimmer231** [comment]
This is now approved by me, I gave the revolver sprites a bit of polish.

**Comment by Rohesie** [comment]
Still has code changes pending.

**Pull request closed without merging**